torch/ao/quantization/experimental/linear.py

```diff
@@ -0,0 +1,168 @@
+import torch
+import numpy as np
+
+from torch.nn.quantized.modules.utils import WeightedQuantizedModule
+from torch.ao.quantization.experimental.observer import APoTObserver
+from torch.ao.quantization.experimental.quantizer import quantize_APoT
+
+class LinearAPoT(WeightedQuantizedModule):
+    r"""
+    A quantized linear module with quantized tensor as inputs and outputs
+    to support APoT quantization.
+    We adopt the same interface as `torch.nn.Linear`, see
+    https://pytorch.org/docs/stable/nn.html#torch.nn.Linear for documentation.
+
+    Similar to :class:`~torch.nn.Linear`, attributes will be randomly
+    initialized at module creation time and will be overwritten later
+
+    Attributes:
+        alpha: `alpha` qparam of output Quantized Tensor, type: Tensor
+        gamma: `gamma` qparam of output Quantized Tensor, type: Tensor
+        quantization_levels: `quantization_levels` qparam of output Quantized Tensor, type: Tensor
+        level_indices: `level_indices` qparam of output Quantized Tensor, type: Tensor
+        weight: APoT quantized tensor from weight2quantize
+        weight_transposed: transposed weight tensor, used in linear transformation calculation (y = x * A^T + b)
+    """
+
+    def __init__(self, weight2quantize: torch.Tensor):
+        assert weight2quantize.dim() == 2
+
+        super().__init__()
+
+        # hard code b, k to match uniform quantization: b=4, k=1
+        self.b = 8
```

Comment:
i'm a bit confused. why are we hardcoding these instead of having them as default parameters? your bitshift matrix multiplication should be generic and shouldn't be limited to b = 8 and k = 1...

Reply:
the uniform linear module scale hardcodes scale=1 and zero_point=0 so i was following that, but i can take b and k as input